### ISSUE TITLE
Define basic conformance classes

### ISSUE THREAD

**Issue by dstenger** [opengeospatial/ets-gpkg12#41]
All basic conformance class need to be passed to get an instance certified as GeoPackage compliant by CITE team.
@jyutzler Do you think that all conformance classes are important for certification?

Proposal in https://github.com/opengeospatial/ets-gpkg12/issues/37#issuecomment-363413431:
From http://www.geopackage.org/spec120/#_core and http://www.geopackage.org/spec120/#_options I understand that only the conformance class ```Core``` is mandatory. I think all other conformance classes should be removed from https://github.com/opengeospatial/ets-gpkg12/blob/27084850e436846e6811f363ce76db44d549aa8d/src/main/config/teamengine/config.xml as ```Core``` is the basic conformance class.

**Comment by dstenger** [comment]
Results must also be applied to https://github.com/opengeospatial/ets-gpkg12-nsg.

**Comment by jyutzler** [comment]
This is a matter of OGC policy regarding encoding standards vs. implementation specifications. The following hasn't been vetted by the OAB or anyone else but I'm going to throw it out there.

> GeoPackage is an Encoding Standard. The Encoding Standard describes the mandatory elements and the available options (which can be considered conformance classes). If a GeoPackage violates any of those conformance classes then it isn't complaint.

> A compliant client of an Encoding Standard is one that behaves appropriately when loading compliant content. That means doing something useful with supported conformance classes and ignoring unsupported conformance classes. Client tests should load a variety of content into the software being evaluated and demonstrate that the software behaves appropriately on all of it. A vendor that declares its client software is compliant with an Encoding Standard should also declare which conformance classes the software handles in a useful way.

> A compliant producer of an encoding standard is one that produces compliant content. Producer tests should produce a set of content covering the gamut of supported conformance classes and verify that the outputs are all compliant. A vendor that declares its producer software is compliant with an Encoding Standard should also declare which conformance classes the software is capable of producing. 

I'm going to bring this up with the OAB in the interest of getting a formal policy we can follow.

**Comment by dstenger** [comment]
Thank you for your input.

However, if we take a look at other specifications, conformance classes can also be implemented "not the standard way" and the service is still compliant to the basic conformance class.

Example: We take a look at WFS 2.0 and the optional Transactional WFS conformance class. A WFS 2.0 can be transactional by not following the standard 100% (tests of Transactional WFS conformance class fail). Still, this WFS can be compliant to the core requirements of specification by passing all Simple WFS requirements. So, this WFS has basic compliance although the transaction functionality is not implemented as required by the standard.

Wouldn't this example also apply to GeoPackage? -> E.g., the core of a GeoPackage is correct but an extension has aspects not following the standard.

@bermud What do you think about this?

**Comment by jyutzler** [comment]
I have opened up an [OAB ticket](https://portal.opengeospatial.org/?m=projects&a=view&project_id=228&tab=5&act=details&issue_id=1227) and the OAB has agreed to address the issue on March 13.

**Comment by lgoltz** [comment]
@jyutzler Can you summarize the results of the telco in relation to this ticket? 

We need to find a solution for this ticket as soon as possible as the ETS should be deployed on production soon.

**Comment by jyutzler** [comment]
Unfortunately the OAB response did not address this aspect of the issue. I think for now there is nothing we can do other than treat all parts of the standard as core. You can't indiscriminately violate any portion of the standard so it is all-or-nothing.
